File: CodeLocatorApp/CodeLocatorCore/src/main/java/com/bytedance/tools/codelocator/operate/ApplicationOperate.java
package com.bytedance.tools.codelocator.operate;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.bytedance.tools.codelocator.CodeLocator;
import com.bytedance.tools.codelocator.model.OperateData;
import com.bytedance.tools.codelocator.model.ResultData;
import com.bytedance.tools.codelocator.utils.ActionUtils;
import com.bytedance.tools.codelocator.utils.CodeLocatorConstants;

/**
 * Created by liujian.android on 2024/4/2
 *
 * @author liujian.android@bytedance.com
 */
public class ApplicationOperate extends Operate {

    @NonNull
    @Override
    public String getOperateType() {
        return CodeLocatorConstants.OperateType.APPLICATION;
    }

    @Override
    public boolean executeCommandOperate(@NonNull Activity activity, @NonNull OperateData operateData, @NonNull ResultData result) {
        for (int i = 0; i < operateData.getDataList().size(); i++) {
            ActionUtils.changeApplicationByAction(
                    CodeLocator.sApplication,
                    activity,
                    operateData.getDataList().get(i),
                    result
            );
        }
        return true;
    }

}

File: CodeLocatorApp/CodeLocatorCore/src/main/java/com/bytedance/tools/codelocator/operate/ApplicationOperate.kt
package com.bytedance.tools.codelocator.operate

import android.app.Activity
import com.bytedance.tools.codelocator.CodeLocator
import com.bytedance.tools.codelocator.model.OperateData
import com.bytedance.tools.codelocator.model.ResultData
import com.bytedance.tools.codelocator.utils.ActionUtils
import com.bytedance.tools.codelocator.utils.CodeLocatorConstants

class ApplicationOperate : Operate() {

    override fun getOperateType(): String = CodeLocatorConstants.OperateType.APPLICATION

    override fun excuteCommandOperate(activity: Activity, operateData: OperateData): Boolean {
        return false
    }

    override fun excuteCommandOperate(
        activity: Activity,
        operateData: OperateData,
        result: ResultData
    ): Boolean {
        for (i in 0 until operateData.dataList.size) {
            ActionUtils.changeApplicationByAction(
                CodeLocator.sApplication,
                activity,
                operateData.dataList[i],
                result
            )
        }
        return true
    }

}